module/zfs/ldi_iokit.cpp

```diff
@@ -1193,20 +1193,34 @@ ldi_iokit_io_intr(void *target, void *parameter,
 	/* In debug builds, verify buffer pointers */
 	ASSERT3U(lbp, !=, 0);
 	ASSERT3U(iobp, !=, 0);
+
+	if (!iobp || !lbp) {
+		printf("%s missing a buffer\n", __func__);
+		return;
+	}
+
 	ASSERT3U(iobp->iomem, !=, 0);
+
+	if (!iobp->iomem) {
+	  printf("%s missing iobp->iomem\n", __func__);
+	  return;
+	}
+
+	ASSERT3U(ldi_zfs_handle, !=, 0);
+
 	if (actualByteCount == 0 ||
 	    actualByteCount != lbp->b_bcount ||
 	    status != kIOReturnSuccess) {
-		dprintf("%s %s %llx / %llx\n", __func__,
+		printf("%s %s %llx / %llx\n", __func__,
 		    "actualByteCount != lbp->b_bcount",
 		    actualByteCount, lbp->b_bcount);
-		dprintf("%s status %d %d %s\n", __func__, status,
-		    ldi_zfs_handle->errnoFromReturn(status),
-		    ldi_zfs_handle->stringFromReturn(status));
-	}
-	if (!iobp || !lbp) {
-		dprintf("%s missing a buffer\n", __func__);
-		return;
+		if (ldi_zfs_handle)
+		  printf("%s status %d %d %s\n", __func__, status,
+			 ldi_zfs_handle->errnoFromReturn(status),
+			 ldi_zfs_handle->stringFromReturn(status));
+		else
+		  printf("%s status %d ldi_zfs_handle is NULL\n",
```

Comment:
I think these prints are in #ifdef DEBUG, but wont it get noisy to print each IO completion?